### PR TITLE
[win32] Fixed CMake configuration and build on Windows

### PR DESCRIPTION
**Pull request by Karlson2k**
If Python is installed by Visual Studio then CMake found it and use it instead of Python from Kodi dependencies.
The main problem is the string `pragma comment(lib,"python37_d.lib")` in standard python header (commented out in the header in Kodi dependencies). As the result, linker failed to build kodi executable.

This situation has been fixed by instructing CMake to search for libraries and headers in Kodi dependencies before checking system location. This may fix other possible problems when some specific library is already installed into the build system on Windows and can be accidentally used instead of Kodi supplied dependencies.

### CODE REVIEW

**Comment by DaveTBlake** [comment]
I am remembering that we had problems with Python version and Windows builds in the past that @Paxxi resolved (or didn't??). @Paxxi if you are about could you review this.

**Comment by Paxxi** [comment]
I can't remember but I know that I failed to fix it for some cases. Using prepend looks like a perfect solution to me

**Review by Paxxi — APPROVED** [review]
Looks good to me

File: cmake/scripts/windows/ArchSetup.cmake [hunk 1]
```diff
@@ -32,9 +32,11 @@ set(DEPS_FOLDER_RELATIVE project/BuildDependencies)
 set(DEPENDENCIES_DIR ${CMAKE_SOURCE_DIR}/${DEPS_FOLDER_RELATIVE}/${ARCH})
 set(MINGW_LIBS_DIR ${CMAKE_SOURCE_DIR}/${DEPS_FOLDER_RELATIVE}/mingwlibs/${ARCH})
 
+# mingw libs
 list(APPEND CMAKE_PREFIX_PATH ${MINGW_LIBS_DIR})
 list(APPEND CMAKE_LIBRARY_PATH ${MINGW_LIBS_DIR}/bin)
-list(APPEND CMAKE_PREFIX_PATH ${DEPENDENCIES_DIR})
+# dependencies
+list(PREPEND CMAKE_PREFIX_PATH ${DEPENDENCIES_DIR})
```
Comment (afedchin) [inline cmake/scripts/windows/ArchSetup.cmake:39]: @Karlson2k `PREPEND` list command was introduced in cmake 3.15 we have to change requirement of cmake minimal version https://github.com/xbmc/xbmc/blob/master/CMakeLists.txt#L1

File: cmake/scripts/windowsstore/ArchSetup.cmake [hunk 2]
```diff
@@ -52,7 +52,7 @@ set(MINGW_LIBS_DIR ${CMAKE_SOURCE_DIR}/${DEPS_FOLDER_RELATIVE}/mingwlibs/win10-$
 list(APPEND CMAKE_PREFIX_PATH ${MINGW_LIBS_DIR})
 list(APPEND CMAKE_LIBRARY_PATH ${MINGW_LIBS_DIR}/bin)
 # dependencies
-list(APPEND CMAKE_PREFIX_PATH ${DEPENDENCIES_DIR})
+list(PREPEND CMAKE_PREFIX_PATH ${DEPENDENCIES_DIR})
```
Comment (afedchin) [inline cmake/scripts/windowsstore/ArchSetup.cmake:55]: same as above

**Pull request merged**